// angular-user-idle/esm2020/lib/angular-user-idle.module.mjs
import { NgModule } from '@angular/core';
import { provideUserIdleConfig } from './angular-user-idle.config';
import * as i0 from "@angular/core";
/**
 * User's idle module.
 * @deprecated since version 4.0.0  - use provideUserIdleConfig(config: UserIdleConfig) instead
 */
export class UserIdleModule {
    static forRoot(config) {
        return {
            ngModule: UserIdleModule,
            providers: [provideUserIdleConfig(config)],
        };
    }
}
UserIdleModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.3", ngImport: i0, type: UserIdleModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
UserIdleModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.1.3", ngImport: i0, type: UserIdleModule });
UserIdleModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.1.3", ngImport: i0, type: UserIdleModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.3", ngImport: i0, type: UserIdleModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [],
                }]
        }] });
